### PR TITLE
Add NXP security disclaimer

### PR DESCRIPTION
**Pull request by clementfaure**
Hello @jbech-linaro @ruchi393 @jenswi-linaro @jforissier 

In this pull-request:

- I added a dedicated space for platform specific documentation. Please let me know if you have a something else in mind regarding the doc structure.
- I added the NXP security disclaimer to the NXP page.

Note: I had to force `docutils<0.18` in `requirements.txt` because of this issue : https://github.com/readthedocs/readthedocs.org/issues/8618. Let me know if you want me to merge this (temporary) fix.

Thanks!
Clement

### CODE REVIEW

**Comment by jbech-linaro** [comment]
Hi @clementfaure , this is looking good to me.
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: architecture/platforms/index.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.. _platform_documentation:
+
+###########################
+Platform documentation
+###########################
```
Comment (etienne-lms) [inline architecture/platforms/index.rst:5]:
```patch
-###########################
+######################
 Platform documentation
+######################
-###########################

File: architecture/porting_guidelines.rst [hunk 2]
```diff
@@ -411,7 +411,7 @@ product. There are a couple of reasons for that.
 
 Because of this we always urge companies and device manufacturers making the end
 product to follow the security guidelines from the chipmaker they are basing
-their products on.
+their products on. Please refer to :ref:`platform_documentation`
```
Comment (etienne-lms) [inline architecture/porting_guidelines.rst:414]: I would rephrase to `Refer also to ...` since that section does not cover all supported platforms, and is likely to do so in the future.

File: architecture/platforms/nxp.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
+.. _nxp:
+
+##################
+NXP
+##################
```
Comment (etienne-lms) [inline architecture/platforms/nxp.rst:5]:
```
###
NXP
###

File: architecture/platforms/index.rst (hunk 1 above)
Comment (clementfaure) [reply architecture/platforms/index.rst:5]: Ok

File: architecture/porting_guidelines.rst (hunk 2 above)
Comment (clementfaure) [reply architecture/porting_guidelines.rst:414]: Ok

File: architecture/platforms/nxp.rst (hunk 3 above)
Comment (clementfaure) [reply architecture/platforms/nxp.rst:5]: Ok

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by ruchi393** [comment]
`Acked-by: Ruchika Gupta <ruchika.gupta@linaro.org>`

**Comment by Emantor** [comment]
Looks great.

`Reviewed-by: Rouven Czerwinski <r.czerwinski@pengutronix.de>`

**Comment by clementfaure** [comment]
@etienne-lms Can I get your `Reviewed-by ..` ?

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` for both commits.

**Comment by clementfaure** [comment]
Tags applied, thank you everyone!
